numpy/core/numeric.py

```diff
@@ -2990,7 +2994,7 @@ class _unspecified(object):
 
 
 @set_module('numpy')
-class errstate(object):
+class errstate(_errstate_baseclass):
```

Comment:
Right now, `_errstate_baseclass` exists only because of python2; should this be the first file where we just remove the python2 work-arounds?

Reply:
Are you suggesting I do this or not?

Reply:
Hmm, yes, I am, but am not sure @charris would agree this is a good way to proceed.

Certainly, it is fine *not* to do this right now.

Reply:
We will not immediately remove workarounds from existing code, but new code doesn't need them unless it is planned for a backport.

Reply:
@charris so should this workaround be removed?

Reply:
@charris - here, there is a specific work-around, as the whole PR doesn't work on python3, but removing it requires also removing other python2 work-arounds. My sense is that we might as well get started (if @nimish is willing), but we can also take the easy route and just go with what is here.

Reply:
What is wrong with just using `ContextDecorator`

> as the whole PR doesn't work on python3

I thought this work was all Python3 only?

Reply:
Indeed, it is, but right now up at the top of the file there are separate paths for imports for python3 and python2 (with `_errstate_baseclass` set to `object` in python2, since the `errstate` class itself works fine on python2. So, the specific question is indeed: should we just use `ContextDecorator` here, and, since that will break usage on python2, also remove the python2-specific path at the top.

Reply:
I'd just import `ContextDecorator` outside of the if statement and leave it alone otherwise. We are keeping the Python 2 workarounds to simplify backports, but I don't think we need to keep them current :)